data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```
@@ -24,15 +42,96 @@
+      }
+    }
+    LOG.info("Consumer shutdown successfully...");
+  }
+
+  private Properties getConsumerProperties(TopicsConfig topicConfig) {
+    Properties properties = new Properties();
+    try {
+      properties.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoCommitInterval().toSecondsPart());
+      properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoOffsetReset());
+      properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, sourceConfig.getBootStrapServers());
+      properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG,
+              topicConfig.getTopic().getConsumerGroupConfig().getAutoCommit());
+      properties.put(ConsumerConfig.GROUP_ID_CONFIG, topicConfig.getTopic().getConsumerGroupConfig().getGroupId());
+      setPropertiesForSchemaType(topicConfig, properties, topicConfig.getTopic().getSchemaConfig().getSchemaType());
+    } catch (Exception e) {
+      LOG.error("Unable to create the Kafka Consumer from the given configurations.", e);
+    }
+
+    return properties;
   }
 
+  private void setPropertiesForSchemaType(TopicsConfig topicConfig, Properties properties, String schemaType) {
+
+    if (schemaType.equalsIgnoreCase(MessageFormat.JSON.toString())) {
+      properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+              topicConfig.getTopic().getSchemaConfig().getKeyDeserializer());
+      properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+    } else if (schemaType.equalsIgnoreCase(MessageFormat.PLAINTEXT.toString())) {
+      properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+              topicConfig.getTopic().getSchemaConfig().getKeyDeserializer());
+      properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+              topicConfig.getTopic().getSchemaConfig().getValueDeserializer());
+    }
```

else? if it's not Json/Plaintext should we throw an exception? also would be good to add a log message on schema type

---

Log message added.
No need of throwing exception as the default type is JSON.